spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ConvLSTMPeephole.scala

```diff
@@ -43,35 +43,37 @@ import scala.reflect.ClassTag
  * @param withPeephole: whether use last cell status control a gate.
  */
 class ConvLSTMPeephole[T : ClassTag] (
-   val inputSize: Int,
-   val hiddenSize: Int,
-   val kernelI: Int,
-   val kernelC: Int,
-   val stride: Int,
-   var wRegularizer: Regularizer[T] = null,
-   var uRegularizer: Regularizer[T] = null,
-   var bRegularizer: Regularizer[T] = null,
-   val withPeephole: Boolean = true
+  val inputSize: Int,
+  val outputSize: Int,
+  val kernelI: Int,
+  val kernelC: Int,
+  val stride: Int,
+  var wRegularizer: Regularizer[T] = null,
+  var uRegularizer: Regularizer[T] = null,
+  var bRegularizer: Regularizer[T] = null,
+  val withPeephole: Boolean = true
 )(implicit ev: TensorNumeric[T])
   extends Cell[T](
-    hiddensShape = Array(hiddenSize, hiddenSize),
+    hiddensShape = Array(outputSize, outputSize),
     regularizers = Array(wRegularizer, uRegularizer, bRegularizer)
   ) {
   var inputGate: Sequential[T] = _
   var forgetGate: Sequential[T] = _
   var outputGate: Sequential[T] = _
   var hiddenLayer: Sequential[T] = _
   var cellLayer: Sequential[T] = _
+  val joinDim = 2
   override var cell: AbstractModule[Activity, Activity, T] = buildConvLSTM()
 
-  def buildGate(): Sequential[T] = {
-    val i2g = Sequential()
-      .add(Contiguous())
-      .add(SpatialConvolution(inputSize, hiddenSize, kernelI, kernelI,
-      stride, stride, kernelI/2, kernelI/2, wRegularizer = wRegularizer,
-        bRegularizer = bRegularizer))
+  override def preTopology: AbstractModule[Activity, Activity, T] =
+    Sequential()
+        .add(TimeDistributed(SpatialConvolution(inputSize, outputSize*4, kernelI, kernelI,
+          stride, stride, kernelI/2, kernelI/2, wRegularizer = wRegularizer,
+          bRegularizer = bRegularizer)))
 
```

Comment:
I add a   `override def getPreTopologyNOutput: Int` in Cell. It is used to insert a BatchNormalization layer between preTopology and topology. If preTopology returns a Sequential, please all insert this line:

`  override def getPreTopologyNOutput: Int = 4 * outputSize`